sonic_data_client/show_client.go

```diff
@@ -0,0 +1,210 @@
+package client
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/Workiva/go-datastructures/queue"
+	log "github.com/golang/glog"
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+	spb "github.com/sonic-net/sonic-gnmi/proto"
+)
+
+type DataGetter func() ([]byte, error)
+
+var showTrie *Trie = NewTrie()
+
+func RegisterCliPath(path []string, getter DataGetter) {
+	n := showTrie.Add(path, getter)
+	if n.meta.(DataGetter) == nil {
+		log.V(1).Infof("Failed to add trie node for %v with %v", path, getter)
+	} else {
+		log.V(2).Infof("Add trie node for %v with %v", path, getter)
+	}
+}
+
+type ShowClient struct {
+	prefix      *gnmipb.Path
+	path2Getter map[*gnmipb.Path]DataGetter
+
+	q       *queue.PriorityQueue
+	channel chan struct{}
+
+	synced sync.WaitGroup  // Control when to send gNMI sync_response
+	w      *sync.WaitGroup // wait for all sub go routines to finish
+	mu     sync.RWMutex    // Mutex for data protection among routines for DbClient
+
+	sendMsg int64
+	recvMsg int64
+	errors  int64
+}
+
+func lookupDataGetter(prefix, path *gnmipb.Path) (DataGetter, error) {
+	stringSlice := []string{prefix.GetTarget()}
+	fullPath := gnmiFullPath(prefix, path)
+
+	elems := fullPath.GetElem()
+	if elems != nil {
+		for i, elem := range elems {
+			log.V(6).Infof("index %d elem : %#v %#v", i, elem.GetName(), elem.GetKey())
+			stringSlice = append(stringSlice, elem.GetName())
+		}
+	}
+	n, ok := showTrie.Find(stringSlice)
+	if ok {
+		getter := n.meta.(DataGetter)
+		return getter, nil
+	}
+	return nil, fmt.Errorf("%v not found in clientTrie tree", stringSlice)
+}
+
+func NewShowClient(paths []*gnmipb.Path, prefix *gnmipb.Path) (Client, error) {
+	var showClient ShowClient
+	showClient.path2Getter = make(map[*gnmipb.Path]DataGetter)
+	showClient.prefix = prefix
+	for _, path := range paths {
+		getter, err := lookupDataGetter(prefix, path)
+		if err != nil {
+			return nil, err
+		}
+		showClient.path2Getter[path] = getter
+	}
+
+	return &showClient, nil
+}
+
+// String returns the target the client is querying.
+func (c *ShowClient) String() string {
+	return fmt.Sprintf("ShowClient Prefix %v  sendMsg %v, recvMsg %v",
+		c.prefix.GetTarget(), c.sendMsg, c.recvMsg)
+}
+
+func (c *ShowClient) Get(w *sync.WaitGroup) ([]*spb.Value, error) {
+	c.w = w
+
+	var values []*spb.Value
+	ts := time.Now()
+	for gnmiPath, getter := range c.path2Getter {
+		v, err := getter()
+		if err != nil {
+			log.V(3).Infof("GetData error %v for %v", err, v)
+			return nil, err
+		}
+		values = append(values, &spb.Value{
+			Prefix:    c.prefix,
+			Path:      gnmiPath,
+			Timestamp: ts.UnixNano(),
+			Val: &gnmipb.TypedValue{
+				Value: &gnmipb.TypedValue_JsonIetfVal{
+					JsonIetfVal: v,
+				}},
+		})
+	}
+	log.V(6).Infof("Getting #%v", values)
+	log.V(4).Infof("Get done, total time taken: %v ms", int64(time.Since(ts)/time.Millisecond))
+	return values, nil
+}
+
+func GetDataFromFile(fileName string) ([]byte, error) {
```

Comment:
these funs looks like show utiltiies, we should move them to show_client directory, and maybe put all them into show_common.go

same for 
GetDataFromTablePaths
CreateTablePathsFromQueries